fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -162,7 +161,8 @@ public Optional<QueryPredicate> toQueryPredicate(@Nullable final TypeRepository
             if (leftChildCorrelatedTo.isEmpty() && rightChildCorrelatedTo.isEmpty() && typeRepository != null) {
                 return tryBoxSelfAsConstantPredicate(typeRepository);
             }
-            if (rightChildCorrelatedTo.contains(innermostAlias) && !leftChildCorrelatedTo.contains(innermostAlias)) {
+            if (localAliases.containsAll(rightChildCorrelatedTo) &&
```

Comment:
Hmm, so what happens if `localAliases` does not contain all of `right`, and/or none of `left`? You just flip over to the else branch. That creates an asymmetry I do not like. On the other hand I don't have something better for you. Any ideas?